com.playeveryware.eos/Runtime/Core/EOS_SDK_Additions/FileRequestTransferWrapper.cs

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2024 PlayEveryWare
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+namespace PlayEveryWare.EpicOnlineServices
+{
+    using Epic.OnlineServices;
+
+    /// <summary>
+    /// Base implementation for file transfer wrapper classes, used to wrap the
+    /// various file transfer request handles that the EOS SDK provides.
+    /// </summary>
+    /// <typeparam name="T">
+    /// Interface defining what functionality a wrapper should define.
+    /// </typeparam>
+    public abstract class FileRequestTransferWrapper<T> : IFileTransferRequest
+        where T : class
+    {
+        /// <summary>
+        /// Wrapped instance.
+        /// </summary>
+        protected T _instance;
+
+        /// <summary>
+        /// Creates a new instance of the FileRequestTransferWrapper.
+        /// </summary>
+        /// <param name="instance">
+        /// The class being wrapped.
+        /// </param>
+        protected FileRequestTransferWrapper(T instance)
+        {
+            _instance = instance;
+        }
+
+        #region Equality operator overloads
+
+        /*
+         * The following section contains methods that are used to seemlessly
+         * provide wrapper functionality for the classes that this class wraps,
+         * allowing the wrapping to be transparent when assigning values to
+         * instances of the wrapper, or when comparing the instance to null (for
+         * instance in scenarios where the instance being wrapped is null, but
+         * the wrapper is not null - in those cases we still want to say it's
+         * equal to null to be as transparent of a wrapper as possible).
+         */
+
+        public static bool operator ==(FileRequestTransferWrapper<T> wrapper, object obj)
+        {
+            if (wrapper is null || wrapper._instance is null)
```

Comment:
I didn't know you could ask if something "is null", I've always used the equals. That's pretty neat.

Reply:
Right? Cool stuff.